example_packages/preprocess_hello_dependency/src/preprocess_hello_dependency.f90

```diff
@@ -0,0 +1,16 @@
+module preprocess_hello_dependency
+  implicit none
+  private
+
+  public :: say_hello
+contains
+  subroutine say_hello
+
+!> This build should break because we do not have FOO defined as macro in the fpm.toml file of
+!> this package.
+#ifndef FOO
```

Comment:
If you use `#ifdef FOO` the package would pass the build and we can safely run it as check in the CI.

Reply:
Yes, added the requested the changes.